sdk/core/azure_core_amqp/src/fe2o3/session.rs

```diff
@@ -98,6 +100,14 @@ impl AmqpSessionApis for Fe2o3AmqpSession {
     }
 
     async fn end(&self) -> Result<()> {
-        todo!()
+        self.session
+            .get()
+            .unwrap()
```

Comment:
Don't unwrap in production code. You're already returning a `Result` so just map the error as appropriate e.g.,

```rust
self.session
    .get()
    .map(|err| Error::new(ErrorKind::Io, err)?
    .lock()
    // ...
}

Reply:
oh, *THAT'S* how to do it (FWIW, it's `ok_or_else`, not `map`).

Reply:
Updated almost all the .unwrap calls.

Any suggestions about how to deal with .unwrap in Into<> and From<>? 

Do I need to use TryFrom instead?

Reply:
As discussed offline, yeah, use `TryFrom` and friends.